action/dock.action

```diff
@@ -0,0 +1,8 @@
+#goal
+bool dock
+---
+#result
+bool docked
+---
+#feedback
+string docking_state
```

Comment:
I let the docking state be a string in the code. But for the message, I think it should be an enum. 

We can backport the state machine code to use enum states later but the new interfaces we add should be proper. What are your thoughts @dean4ta @khrisnakamarga ?